edb/edgeql/compiler/stmt.py

```diff
@@ -254,16 +258,105 @@ def compile_GroupQuery(
     return result
 
 
-@dispatch.compile.register(qlast.InsertQuery)
-def compile_InsertQuery(
-        expr: qlast.InsertQuery, *, ctx: context.ContextLevel) -> irast.Set:
+def simple_stmt_eq(lhs: irast.Base, rhs: irast.Base) -> bool:
+    if (
+        isinstance(lhs, irast.BaseConstant)
+        and isinstance(rhs, irast.BaseConstant)
+        and lhs.typeref == rhs.typeref  # XXX?
```

Comment:
How about using `evaluate_to_python_val` here to compute the value of the constant for comparison?

Reply:
Oh, yeah. I looked at doing that for the whole comparison and didn't because it wouldn't work for parameters but it should work here.

Does the typeref `==` comparison actually work, by the way?

Reply:
It does so by accident because we maintain a `type -> typeref` cache.  If you want to compare them reliably, you have to compare `typeref.id`, but it'd be much better to do `ir_typeref_to_type` and compare the type object instead.